BCStudentSoftwareDevTeam/celts
# Celts kiosk mode issue#51

This PR fixes issue #51 

Summary:
Created a Kiosk page for the CELTS team that will allow them to quickly scan in participants who have shown up at an event. In order to accomplish this, we built a plain page with a header that will display the Current event you signing in for and a Text box that will auto focus on page load and when you toggle full screen. The participant will then scan their badge and the Kiosk page will send their scan code to the back end where we parse the scan and pull out the B number and toggle the user as attended in the database and then upon success flash a message displaying the user first and last name and a successful sign in. We also added a flasher that will display if a user is already signed in to the event. In case a participant forgets their id badge upon arriving at the event the user can manually enter in their B number and press enter and the same steps will be completed. 

To Test:
We have added functions to the test suite that will pass if ran. You can also load up the Kiosk page at this URL: `http://<YOURIP>/1/kiosk` and use a scanner to scan your Id badge or manually type in your Id number. If you are in the system as an eligible participant signed up for this event you will be signed in. If you attempt twice you will be told you are already signed in. If any error occurs you will be flashed a message to see the attendant.

Notes: By default, the browser has priority over the `esc` button. The way we have designed the Kiosk page when clicking the provided toggle the page will enter and exit full screen and hide any elements not necessary for a Participant to use. Clicking `esc` will exit fullscreen but will not bring back the navbar and the footer. Clicking a second time will complete this action.
